### PR TITLE
687 javascipt progress window

### PR DESCRIPTION
**Pull request by nnunn**

### CODE REVIEW

**Comment by pbinkley** [comment]
Could you remove the .DS_Store files? And we should add that to .gitignore.

**Comment by nnunn** [comment]
done - sorry about that

File: app/views/my/index.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class="lightbox" id="lightbox">
   <div class="lightbox-content">
-    <a href="javascript:void(0)" onclick="document.getElementById('lightbox').style.display='none';" class="textright"><span class="glyphicon glyphicon-remove my-close"></span></a>
+    <a href="https://hydranorthdev.library.ualberta.ca/dashboard/files?closed=yes" class="textright"><span class="glyphicon glyphicon-remove my-close"></span></a>
```
Comment (pgwillia) [inline app/views/my/index.html.erb]: I'm guessing this won't work outside of hydranorthdev?

File: app/views/my/index.html.erb [hunk 2]
```diff
@@ -11,7 +11,7 @@
 </div>
       <h2>Thank you for adding your research to ERA!</h2>
       <p> We are currently processing your item(s). This may take awhile.</p>
-      <p> <strong>If you would like to continue using ERA while we process, you can <a href="javascript:void(0)" onclick="document.getElementById('lightbox').style.display='none';" class="close-button">close</a> this window at any time. </strong></p>
+      <p> <strong>If you would like to continue using ERA while we process, you can <a href="https://hydranorthdev.library.ualberta.ca/dashboard/files?closed=yes">close</a> this window at any time. </strong></p>
```
Comment (pgwillia) [inline app/views/my/index.html.erb]: Same as above.

**Comment by nnunn** [comment]
fixed to relative links - thanks for catching that.

**Comment by pgwillia** [comment]
Working out conflicts with sprint_27 branch and looking into failing tests.

**Comment by pgwillia** [comment]
fixed conflicts.  looking at spec failures.

**Comment by pgwillia** [comment]
Failures:

  1) RedirectController#item Item should be migrated
     Failure/Error: expect(subject.fedora3uuid).to eq "uuid:394266f0-0e4a-42e6-a199-158165226426"

       expected: "uuid:394266f0-0e4a-42e6-a199-158165226426"
            got: nil

       (compared using ==)
     # ./spec/controllers/redirect_controller_spec.rb:37:in `block (3 levels) in <top (required)>'

  2) RedirectController#datastream Item should be migrated
     Failure/Error: expect(subject.fedora3uuid).to eq "uuid:394266f0-0e4a-42e6-a199-158165226426"

       expected: "uuid:394266f0-0e4a-42e6-a199-158165226426"
            got: nil

       (compared using ==)
     # ./spec/controllers/redirect_controller_spec.rb:67:in `block (3 levels) in <top (required)>'

  3) SAML Users with no existing account should be able to use SAML to create acount
     Failure/Error: expect(page).to have_content "Do you have an existing ERA account?"
       expected to find text "Do you have an existing ERA account?" in "SyntaxError at /users/myself@testshib-dot-org/link_account ========================================================== > syntax error, unexpected ':', expecting ')' ...ppend=( submit_tag 'Continue' :class => \"btn btn-primary\" );... ... ^ app/views/users/link_account.html.erb, line 9 --------------------------------------------- ``` ruby 4 5 6 7 If YES, please enter the email you used to register for an ERA Account and follow the instructions emailed to you: 8 > 9 \"btn btn-primary\" %> 10 ``` App backtrace ------------- - app/views/users/link_account.html.erb:9:in `' - spec/features/saml_spec.rb:161:in `block in sign_in_with_saml' - spec/features/saml_spec.rb:161:in `sign_in_with_saml' - spec/features/saml_spec.rb:24:in `block (3 levels) in ' 

  4) SAML Users with no existing account should be redirected to their protected target after creating an account
     Failure/Error: expect(page).to have_content "Do you have an existing ERA account?"
       expected to find text "Do you have an existing ERA account?" in "SyntaxError at /users/myself@testshib-dot-org/link_account ========================================================== > syntax error, unexpected ':', expecting ')' ...ppend=( submit_tag 'Continue' :class => \"btn btn-primary\" );... ... ^ app/views/users/link_account.html.erb, line 9 --------------------------------------------- ``` ruby 4 5 6 7 If YES, please enter the email you used to register for an ERA Account and follow the instructions emailed to you: 8 > 9 \"btn btn-primary\" %> 10 ``` App backtrace ------------- - app/views/users/link_account.html.erb:9:in `' - spec/features/saml_spec.rb:32:in `block (4 levels) in ' - spec/features/saml_spec.rb:32:in `block (3 levels) in ' 

  5) SAML Users with existing legacy accounts logging in via SAML for the first time should receive an email confirmation to link to their existing account
     Failure/Error: expect(page).to have_content "Do you have an existing ERA account?"
       expected to find text "Do you have an existing ERA account?" in "SyntaxError at /users/myself@testshib-dot-org/link_account ========================================================== > syntax error, unexpected ':', expecting ')' ...ppend=( submit_tag 'Continue' :class => \"btn btn-primary\" );... ... ^ app/views/users/link_account.html.erb, line 9 --------------------------------------------- ``` ruby 4 5 6 7 If YES, please enter the email you used to register for an ERA Account and follow the instructions emailed to you: 8 > 9 \"btn btn-primary\" %> 10 ``` App backtrace ------------- - app/views/users/link_account.html.erb:9:in `' - spec/features/saml_spec.rb:161:in `block in sign_in_with_saml' - spec/features/saml_spec.rb:161:in `sign_in_with_saml' - spec/features/saml_spec.rb:66:in `block (4 levels) in ' 

  6) SAML Users with existing legacy accounts logging in via SAML for the first time should not be able to step out of the linking workflow
     Failure/Error: expect(page).to have_content "Do you have an existing ERA account?"
       expected to find text "Do you have an existing ERA account?" in "SyntaxError at /users/myself@testshib-dot-org/link_account ========================================================== > syntax error, unexpected ':', expecting ')' ...ppend=( submit_tag 'Continue' :class => \"btn btn-primary\" );... ... ^ app/views/users/link_account.html.erb, line 9 --------------------------------------------- ``` ruby 4 5 6 7 If YES, please enter the email you used to register for an ERA Account and follow the instructions emailed to you: 8 > 9 \"btn btn-primary\" %> 10 ``` App backtrace ------------- - app/views/users/link_account.html.erb:9:in `' - spec/features/saml_spec.rb:161:in `block in sign_in_with_saml' - spec/features/saml_spec.rb:161:in `sign_in_with_saml' - spec/features/saml_spec.rb:92:in `block (4 levels) in ' 

  7) SAML Users with existing legacy accounts logging in via SAML for the first time should not be able to enumerate user accounts via the linking workflow
     Failure/Error: expect(page).to have_content "Do you have an existing ERA account?"
       expected to find text "Do you have an existing ERA account?" in "SyntaxError at /users/myself@testshib-dot-org/link_account ========================================================== > syntax error, unexpected ':', expecting ')' ...ppend=( submit_tag 'Continue' :class => \"btn btn-primary\" );... ... ^ app/views/users/link_account.html.erb, line 9 --------------------------------------------- ``` ruby 4 5 6 7 If YES, please enter the email you used to register for an ERA Account and follow the instructions emailed to you: 8 > 9 \"btn btn-primary\" %> 10 ``` App backtrace ------------- - app/views/users/link_account.html.erb:9:in `' - spec/features/saml_spec.rb:161:in `block in sign_in_with_saml' - spec/features/saml_spec.rb:161:in `sign_in_with_saml' - spec/features/saml_spec.rb:101:in `block (4 levels) in '

  8) SAML Users with existing legacy accounts who log in with their legacy credentials should not be prompted to confirm their email addresses when linking a CCID
     Failure/Error: expect(ActionMailer::Base.deliveries).to be_empty
       expected `[#<Mail::Message:96872160, Multipart: false, Headers: <Date: Tue, 13 Oct 2015 11:54:30 -0600>, <From: hydranorth@ualberta.ca>, <Reply-To: hydranorth@ualberta.ca>, <To: user64@example.com>, <Message-ID: <561d45564be3e_12359cb3b477345@hydranorthdev.library.ualberta.ca.mail>>, <Subject: Confirmation instructions>, <Mime-Version: 1.0>, <Content-Type: text/html>, <Content-Transfer-Encoding: 7bit>>].empty?` to return true, got false
     # ./spec/features/saml_spec.rb:124:in `block (4 levels) in <top (required)>'

  9) Migration rake tasks migration:dataverse dataverse item should be migrated
     Failure/Error: expect(subject.remote_resource).to eq "dataverse"

       expected: "dataverse"
            got: nil

       (compared using ==)
     # ./spec/tasks/dataverse_migration_rake_spec.rb:28:in `block (3 levels) in <top (required)>'

  10) Migration rake tasks migration:eraitem - multifile item multifile item should have a zip file
      Got 0 failures and 2 other errors:

      10.1) Failure/Error: id = doc["id"]
            NoMethodError:
              undefined method `[]' for nil:NilClass
            # ./spec/tasks/migration_rake_spec.rb:71:in `block (3 levels) in <top (required)>'

      10.2) Failure/Error: @file.delete
            NoMethodError:
              undefined method `delete' for nil:NilClass
            # ./spec/tasks/migration_rake_spec.rb:76:in `block (3 levels) in <top (required)>'

**Comment by pgwillia** [comment]
None of those seem related to this branch.  I'm okay to merge.

**Comment by mbarnett** [comment]
These are, afaik, fixed on sprint 27 itself

**Comment by pgwillia** [comment]
Looking at the SAML failures again.

**Pull request merged**